.github/workflows/haskell.yml

```diff
@@ -112,7 +112,7 @@ jobs:
     # this should be a no-op, but if the cache key was not found we need to build stuff so we can
     # cache it for the next step.
     - name: Install dependencies
-      run: cabal build all --enable-tests --only-dependencies -j --ghc-option=-j4
```

Comment:
Why is the number in the `cabal` flag not specified but it is in the `--ghc-option`? I don't know how many cpus in the GHA runner, but I imagine it is a reasonable setting anyway

Reply:
Github runners have 4 cores. I have no idea why it was not specified for cabal. 
I vaguely remember we had issues when building too many packages concurrently, but I can't reproduce it anymore and `-j4` was already maximum concurrency in GHA so it's kind of redundant.